storage/storage.go

```diff
@@ -587,8 +587,8 @@ func signedURLV4(bucket, name string, opts *SignedURLOptions, now time.Time) (st
 	for k, v := range opts.QueryParameters {
 		canonicalQueryString[k] = append(canonicalQueryString[k], v...)
 	}
-
-	fmt.Fprintf(buf, "%s\n", canonicalQueryString.Encode())
+	escapedQuery := strings.Replace(canonicalQueryString.Encode(), "+", "%20", -1)
```

Comment:
Will you add a comment explaining this? I would expect `Encode` to handle everything.

Reply:
If the value is encoded, then replaced there will be no `+` left to replace right? `+` should be encoded to `%2B`.

Reply:
The default behavior is that `+` is encoded to `%2B` (which is good and expected) but a space is encoded to `+` (while we want `%20` by the spec for this feature). See https://github.com/golang/go/blob/master/src/net/url/url.go#L314 . I'll add a comment here to clarify.

Reply:
Ah, gotcha!